apps/dsaparam.c

```diff
@@ -25,6 +25,8 @@ NON_EMPTY_TRANSLATION_UNIT
 # include <openssl/x509.h>
 # include <openssl/pem.h>
 
+#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
```

Comment:
move to a global file?

Reply:
I agree that it would be better to have a single \#define instead of repeating them in the c-files.

```
genrsa.c:32:#define RSA_RECOMMENDED_MAX_KEY_LENGTH    16384
gendsa.c:28:#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
req.c:47:#define RSA_RECOMMENDED_MAX_KEY_LENGTH    16384
req.c:48:#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
dsaparam.c:28:#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
```
However, currently I don't see an appropriate header file inside the app directory where to place the \#define's. And adding a public \#define seems wrong to me.

@levitte, @rsalz, suggestions?

Reply:
opensslconf.h?

Reply:
DSA limits

https://github.com/openssl/openssl/blob/1e653d0ff7fc2e06a1cb1e05c01feecde84e67d3/include/openssl/dsa.h#L30

Reply:
I'm a little bit confused now: Couldn't you just use the existing constants OPENSSL_XXX_MAX_MODULUS_BITS ... 


```
include/openssl/dsa.h:30:#  define OPENSSL_DSA_MAX_MODULUS_BITS   10000
include/openssl/dh.h:30:#  define OPENSSL_DH_MAX_MODULUS_BITS    10000
include/openssl/rsa.h:31:#  define OPENSSL_RSA_MAX_MODULUS_BITS   16384
```

... instead of \#defining your own?

```
apps/genrsa.c:32:#define RSA_RECOMMENDED_MAX_KEY_LENGTH    16384
apps/gendsa.c:28:#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
apps/req.c:47:#define RSA_RECOMMENDED_MAX_KEY_LENGTH    16384
apps/req.c:48:#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
apps/dsaparam.c:28:#define DSA_RECOMMENDED_MAX_KEY_LENGTH    8192
```

Reply:
Yes, I will take this .h-file values.
Thanks for your hint.